MapboxNavigation/InstructionPresenter.swift

```diff
@@ -63,105 +74,103 @@ class InstructionPresenter {
         return attributedPairs.attributedStrings
     }
     
-    typealias AttributedInstructionComponents = (components: [VisualInstructionComponent], attributedStrings: [NSAttributedString])
+    typealias AttributedInstructionComponents = (components: [VisualInstruction.Component], attributedStrings: [NSAttributedString])
     
     func attributedPairs(for instruction: VisualInstruction, dataSource: DataSource, imageRepository: ImageRepository, onImageDownload: @escaping ImageDownloadCompletion) -> AttributedInstructionComponents {
```

Comment:
The changes to this method cause the compiler to segfault consistently in Xcode 10:

```
1.	While emitting IR SIL function "@$s16MapboxNavigation20InstructionPresenterC15attributedPairs3for10dataSource15imageRepository15onImageDownloadSay0A10Directions06VisualC0C9ComponentOG10components_SaySo18NSAttributedStringCG0E7StringstAK_AA0cd4DataI0_pAA0mK0CySo7UIImageCSgctF".
 for 'attributedPairs(for:dataSource:imageRepository:onImageDownload:)' (at /Users/distiller/project/MapboxNavigation/InstructionPresenter.swift:79:5)
0  swift                    0x0000000110470ee3 PrintStackTraceSignalHandler(void*) + 51
1  swift                    0x00000001104706bc SignalHandler(int) + 348
2  libsystem_platform.dylib 0x00007fff64c1ab5d _sigtramp + 29
3  libsystem_platform.dylib 0x00007ffee39ba650 _sigtramp + 2128214800
4  swift                    0x000000010c44bc73 swift::irgen::SingleScalarTypeInfo<(anonymous namespace)::BridgeObjectTypeInfo, swift::irgen::ReferenceTypeInfo>::copy(swift::irgen::IRGenFunction&, swift::irgen::Explosion&, swift::irgen::Explosion&, swift::irgen::Atomicity) const + 51
5  swift                    0x000000010c486e4e swift::irgen::RecordTypeInfo<(anonymous namespace)::LoadableStructTypeInfo, swift::irgen::LoadableTypeInfo, (anonymous namespace)::StructFieldInfo, true, true>::copy(swift::irgen::IRGenFunction&, swift::irgen::Explosion&, swift::irgen::Explosion&, swift::irgen::Atomicity) const + 94
6  swift                    0x000000010c486e4e swift::irgen::RecordTypeInfo<(anonymous namespace)::LoadableStructTypeInfo, swift::irgen::LoadableTypeInfo, (anonymous namespace)::StructFieldInfo, true, true>::copy(swift::irgen::IRGenFunction&, swift::irgen::Explosion&, swift::irgen::Explosion&, swift::irgen::Atomicity) const + 94
7  swift                    0x000000010c486e4e swift::irgen::RecordTypeInfo<(anonymous namespace)::LoadableStructTypeInfo, swift::irgen::LoadableTypeInfo, (anonymous namespace)::StructFieldInfo, true, true>::copy(swift::irgen::IRGenFunction&, swift::irgen::Explosion&, swift::irgen::Explosion&, swift::irgen::Atomicity) const + 94
8  swift                    0x000000010c486e4e swift::irgen::RecordTypeInfo<(anonymous namespace)::LoadableStructTypeInfo, swift::irgen::LoadableTypeInfo, (anonymous namespace)::StructFieldInfo, true, true>::copy(swift::irgen::IRGenFunction&, swift::irgen::Explosion&, swift::irgen::Explosion&, swift::irgen::Atomicity) const + 94
9  swift                    0x000000010c4d8b45 swift::SILInstructionVisitor<(anonymous namespace)::IRGenSILFunction, void>::visit(swift::SILInstruction*) + 10053
10 swift                    0x000000010c4d3a4a swift::irgen::IRGenModule::emitSILFunction(swift::SILFunction*) + 9898
11 swift                    0x000000010c3915c0 swift::irgen::IRGenerator::emitGlobalTopLevel(bool) + 1232
12 swift                    0x000000010c4b3414 performIRGeneration(swift::IRGenOptions&, swift::ModuleDecl*, std::__1::unique_ptr<swift::SILModule, std::__1::default_delete<swift::SILModule> >, llvm::StringRef, swift::PrimarySpecificPaths const&, llvm::LLVMContext&, swift::SourceFile*, llvm::GlobalVariable**) + 1076
13 swift                    0x000000010c2d5319 performCompile(swift::CompilerInstance&, swift::CompilerInvocation&, llvm::ArrayRef<char const*>, int&, swift::FrontendObserver*, swift::UnifiedStatsReporter*) + 49433
14 swift                    0x000000010c2c592e swift::performFrontend(llvm::ArrayRef<char const*>, char const*, void*, swift::FrontendObserver*) + 6862
15 swift                    0x000000010c2639ce main + 1246
16 libdyld.dylib            0x00007fff64a353d5 start + 1
17 libdyld.dylib            0x00000000000000ef start + 2606542107
error: Segmentation fault: 11
```

Reply:
Looks like it's passing a swift-only type to an obj-c method. Will fix.

Reply:
Maybe we should just drop Xcode 10 support, as in #2282.